tests/unit/containers/my-cooperations/cooperation-details/CooperationDetails.spec.jsx

```diff
@@ -70,42 +80,53 @@ describe('CooperationDetails', () => {
     })
   })
 
-  it('should render details page', () => {
-    const notesButton = screen.getByText('cooperationsPage.details.notes')
+  it('should render details page', async () => {
+    const notesButton = await screen.findByText(
+      'cooperationsPage.details.notes'
+    )
     expect(notesButton).toBeInTheDocument()
   })
 
-  it('should show cooperation status and title', () => {
-    const title = screen.getByText(cooperationMock.title)
-    const statusChip = screen.getByText(cooperationMock.status)
+  it('should show cooperation status and title', async () => {
+    const title = await screen.findByText(cooperationMock.title)
+    const statusChip = await screen.findByText(cooperationMock.status)
 
     expect(title).toBeInTheDocument()
     expect(statusChip).toBeInTheDocument()
   })
 
-  it('should render the component with tabs', () => {
-    const tab1 = screen.getByText('cooperationsPage.tabs.activities')
+  it('should render the component with tabs', async () => {
+    const tab1 = await screen.findByText('cooperationsPage.tabs.activities')
 
     expect(tab1).toBeInTheDocument()
 
-    const tab2 = screen.getByText('cooperationsPage.tabs.details')
-
-    fireEvent.click(tab2)
+    const tab2 = await screen.findByText('cooperationsPage.tabs.details')
 
+    await act(async () => {
```

Comment:
Please, could you explain why do we need to use `act` here?
The same for all the following tests.

Reply:
It is good practice to use act when the state can update, but I don`t understand why you make a callback inside to by async

Reply:
Thank for you advice. I dived diper in this topic and made some changes